### PR TITLE
documentation #798: updated swagger doc for group endpoints

### LINKED ISSUE #798: Update Swagger doc for Group (2)
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Group. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse()
  @ApiBadRequestResponse({
    description: `\n
    InvalidEntityIdException:
      User with such id is not found

    AlreadyRegisteredException:
      User is already registered

    InvalidBodyException:
      Group id can not be empty
      IsCaptain must be a boolean
      IsCaptain can not be empty
    
    AbsenceOfCaptainException:
      Captain was not found
    
    AbsenceOfCaptainTelegramException:
      Captain's telegramId was not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action
    
    ForbiddenException:
      Forbidden`,
  })
  @ApiParam({
    name: 'userId',
    required: true,
    description: 'Id of a user to request to the group',
  })
  @ApiEndpoint({
    summary: 'Request captain or admin to join the group',
    permissions: PERMISSION.USERS_$USERID_GROUP_REQUEST,
  })
  @Patch('/:userId/requestNewGroup')
  requestNewGroup (
    @Param('userId', UserByIdPipe) userId: string,
    @Body() body: GroupRequestDTO,
  ) {
    return this.userService.requestNewGroup(userId, body);
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- api endpoint for summary, permissions, guards

You should also add **field description** in the bady or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

image

**Example**: #777

_My DM is always open for you. Happy coding_ 🐸

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by fujivara**
- DELETE /v2/groups/{groupId}
- GET /v2/groups/{groupId}/students
- GET /v2/groups/{groupId}/captain
- GET /v2/groups/{groupId}/disciplineTeachers
- GET /v2/groups/{groupId}/disciplines

closes #798

### CODE REVIEW

File: src/v2/api/controllers/GroupController.ts [hunk 1]
```diff
@@ -247,7 +264,10 @@ export class GroupController {
   @ApiBadRequestResponse({
     description: `\n
     InvalidGroupIdException: 
-      Group with such id is not found`,
+      Group with such id is not found
+
```
Comment (kropli) [inline src/v2/api/controllers/GroupController.ts:268]: No space is required between exceptions

File: src/v2/api/controllers/GroupController.ts [hunk 2]
```diff
@@ -279,9 +306,9 @@ export class GroupController {
     description: `\n
     InvalidGroupIdException:
       Group with such id is not found
-  
+
```
Comment (kropli) [inline src/v2/api/controllers/GroupController.ts:309]: No space is required between exceptions

**Review by kropli — CHANGES_REQUESTED** [review]

**Review by kropli — APPROVED** [review]

File: src/v2/api/responses/StudentResponse.ts [hunk 3]
```diff
@@ -70,6 +71,7 @@ export class OrdinaryStudentResponse extends BaseStudentResponse {
 
   @ApiProperty({
```
Comment (aim4ik11) [inline src/v2/api/responses/StudentResponse.ts:72]: Remove questionmark near telegramId

File: src/v2/api/controllers/GroupController.ts [hunk 4]
```diff
@@ -189,16 +189,24 @@ export class GroupController {
...
+    description: `\n
+      InvalidEntityIdException: 
+        Group with such id is not found`,
   })
   @ApiForbiddenResponse({
-    description: `
-      NoPermissionException: You do not have permission to perform this action
-    `,
+    description: `\n
+      NoPermissionException:
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:198]: Add unauthorized exception
image

File: src/v2/api/responses/StudentResponse.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export class BaseStudentResponse {
 
   @ApiPropertyOptional({
```
Comment (aim4ik11) [inline src/v2/api/responses/StudentResponse.ts:13]: Change decorator to ApiProperty (make a required field)

File: src/v2/api/controllers/GroupController.ts [hunk 6]
```diff
@@ -247,7 +262,9 @@ export class GroupController {
   @ApiBadRequestResponse({
     description: `\n
     InvalidGroupIdException: 
-      Group with such id is not found`,
+      Group with such id is not found
+    AbsenceOfCaptainException:
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:266]: Place enter between 2 different exceptions

File: src/v2/api/controllers/GroupController.ts [hunk 7]
```diff
@@ -259,7 +276,14 @@ export class GroupController {
     NoPermissionException: 
       You do not have permission to perform this action`,
   })
-  @Access(PERMISSION.GROUPS_$GROUPID_CAPTAIN_GET)
+  @ApiParam({
+    name: 'groupId',
+    description: 'Id of the group to get the captain from',
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:281]: Add required: true

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
In teacherresponse change optional fields with required ones
image

File: src/v2/api/controllers/GroupController.ts [hunk 8]
```diff
@@ -279,7 +303,6 @@ export class GroupController {
     description: `\n
     InvalidGroupIdException:
       Group with such id is not found
-  
     DataMissingException:
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:306]: Add enter between exceptions

**Review by aim4ik11 — CHANGES_REQUESTED** [review]

File: src/v2/api/controllers/GroupController.ts [hunk 9]
```diff
@@ -293,7 +324,14 @@ export class GroupController {
     NoPermissionException: 
       You do not have permission to perform this action`,
   })
-  @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_TEACHERS_GET)
+  @ApiParam({
+    name: 'groupId',
+    description: 'Id of the group to get the discipline teachers from',
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:329]: Add required true

File: src/v2/api/controllers/GroupController.ts [hunk 10]
```diff
@@ -324,7 +362,14 @@ export class GroupController {
     NoPermissionException: 
       You do not have permission to perform this action`,
   })
-  @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_GET)
+  @ApiParam({
+    name: 'groupId',
+    description: 'Id of the group to get the disciplines from',
```
Comment (aim4ik11) [inline src/v2/api/controllers/GroupController.ts:367]: Add required true

**Review by aim4ik11 — CHANGES_REQUESTED** [review]

**Review by aim4ik11 — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**